src/transformers/tokenization_pegasus.py

```diff
@@ -0,0 +1,124 @@
+        # self.added_tokens_decoder = self._added_toks
+        self.decoder: Dict[str, int] = {v: k for k, v in self.encoder.items()}
+        assert 104 in self.encoder
+        assert 105 not in self.encoder
+        assert self.pad_token_id == 0, "pad should be 0"
+        assert self.eos_token_id == 1, "eos should be 1"
+        assert self.unk_token_id != 2
+
+    def _convert_token_to_id(self, token: str) -> int:
+        """ Converts a token (str) in an id using the vocab. """
+
+        if token in self.decoder:
+            return self.decoder[token]
+        elif token in self.added_tokens_decoder:
+            return self.added_tokens_decoder[token]
+        sp_id = self.sp_model.piece_to_id(token)
+        return sp_id + self.offset
+
+    def _convert_id_to_token(self, index: int) -> str:
+        """Converts an index (integer) in a token (str) using the vocab."""
+        if index in self.encoder:
+            return self.encoder[index]
+        elif index in self.added_tokens_encoder:
+            return self.added_tokens_encoder[index]
+        else:
+            # assert index > self.offset, f"cannot decode ids between 2 and {self.offset}. Got {index}"
+            token = self.sp_model.IdToPiece(index - self.offset)
+        return token
+
+    @property
+    def vocab_size(self) -> int:
+        return len(self.sp_model) + self.offset
+
+    def get_vocab(self) -> Dict[str, int]:
+        vocab = {self.convert_ids_to_tokens(i): i for i in range(self.vocab_size)}
+        vocab.update(self.added_tokens_encoder)
+        return vocab
+
+    def num_special_tokens_to_add(self, pair=False):
+        """Just EOS"""
+        return 1
+
+    def _special_token_mask(self, seq):
+        all_special_ids = set(self.all_special_ids)  # call it once instead of inside list comp
+        all_special_ids.remove(self.unk_token_id)  # <unk> is only sometimes special
+        assert all_special_ids == set([0, 1])
+        return [1 if x in all_special_ids else 0 for x in seq]
+
+    def get_special_tokens_mask(
+        self, token_ids_0: List, token_ids_1: Optional[List] = None, already_has_special_tokens: bool = False
+    ) -> List[int]:
+        """Get list where entries are [1] if a token is [eos] or [pad] else 0."""
+        if already_has_special_tokens:
+            return self._special_token_mask(token_ids_0)
+        elif token_ids_1 is None:
+            return self._special_token_mask(token_ids_0) + [1]
+        else:
+            return self._special_token_mask(token_ids_0 + token_ids_1) + [1]
+
+    def build_inputs_with_special_tokens(self, token_ids_0, token_ids_1=None) -> List[int]:
+        """Build model inputs from a sequence by appending eos_token_id."""
+        if token_ids_1 is None:
+            return token_ids_0 + [self.eos_token_id]
+        # We don't expect to process pairs, but leave the pair logic for API consistency
+        return token_ids_0 + token_ids_1 + [self.eos_token_id]
```

Comment:
Would be great to keep the same format as the other models' documentation here, for example [RoBERTa](https://70643-155220641-gh.circle-artifacts.com/0/docs/_build/html/model_doc/roberta.html#transformers.RobertaTokenizer.build_inputs_with_special_tokens)